src/main/java/it/pagopa/transactions/services/TransactionsService.java

```diff
@@ -336,164 +350,195 @@ public Mono<TransactionInfoDto> updateTransactionAuthorization(
         return uuidUtils.uuidFromBase64(transactionId)
                 .fold(
                         Mono::error,
-                        transactionIdDecoded -> Mono.just(transactionIdDecoded)
-                                .doOnNext(el -> log.info("decoded transaction id: {}", transactionIdDecoded))
-                                .flatMap(
-                                        el -> transactionsViewRepository
-                                                .findById(transactionIdDecoded.toString())
-                                                .switchIfEmpty(
-                                                        Mono.error(
-                                                                new TransactionNotFoundException(
-                                                                        transactionIdDecoded.toString()
+                        transactionIdDecoded -> {
+                            TransactionId trnId = new TransactionId(transactionIdDecoded);
+                            log.info("decoded transaction id: {}", transactionIdDecoded);
+                            Mono<BaseTransaction> baseTransaction = transactionsUtils.reduceEvents(trnId);
+                            return wasTransactionAuthorized(trnId)
+                                    .flatMap(alreadyAuthorized -> {
```

Comment:
You can add type annotations here so that you can use just `Either.right` and `Either.left` later:
```suggestion
                                    .<Either<TransactionInfoDto, Mono<BaseTransaction>>>flatMap(alreadyAuthorized -> {
```

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/168/commits/abfdb3480ce8cb9b2d53ad44070cbd6535e71c4c